File: data/EXP01/python/2326/2326.output_1.py
# @lc app=leetcode id=2326 lang=python3
from typing import List, Optional
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def spiralMatrix(self, m: int, n: int, head: Optional[ListNode]) -> List[List[int]]:
        matrix = [[-1] * n for _ in range(m)]
        
        values = []
        node = head
        while node:
            values.append(node.val)
            node = node.next
        
        top, bottom, left, right = 0, m-1, 0, n-1
        row, col = 0, 0
        direction = 0
        
        for i in range(m * n):
            if direction == 0: # Move right
                matrix[row][col] = values[i]
                col += 1
                if col > right:
                    direction = (direction + 1) % 4
                    top += 1
            elif direction == 1: # Move down
                matrix[row][col] = values[i]
                row += 1
                if row > bottom:
                    direction = (direction + 1) % 4
                    right -= 1
            elif direction == 2: # Move left
                matrix[row][col] = values[i]
                col -= 1
                if col < left:
                    direction = (direction + 1) % 4
                    bottom -= 1
            elif direction == 3: # Move up
                matrix[row][col] = values[i]
                row -= 1
                if row < top:
                    direction = (direction + 1) % 4
                    left += 1
        
        return matrix

File: data/EXP01/go/2326/2326.output_1.go
type ListNode struct {
    Val  int
    Next *ListNode
}

func spiralMatrix(m int, n int, head *ListNode) [][]int {
    matrix := make([][]int, m)
    for i := range matrix {
        matrix[i] = make([]int, n)
    }

    nums := make([]int, 0)
    curr := head
    for curr != nil {
        nums = append(nums, curr.Val)
        curr = curr.Next
    }

    left, right := 0, n-1
    top, bottom := 0, m-1
    index := 0

    for left <= right && top <= bottom {
        for j := left; j <= right; j++ {
            if index < len(nums) {
                matrix[top][j] = nums[index]
                index++
            } else {
                matrix[top][j] = -1
            }
        }
        top++

        for i := top; i <= bottom; i++ {
            if index < len(nums) {
                matrix[i][right] = nums[index]
                index++
            } else {
                matrix[i][right] = -1
            }
        }
        right--

        if top <= bottom {
            for j := right; j >= left; j-- {
                if index < len(nums) {
                    matrix[bottom][j] = nums[index]
                    index++
                } else {
                    matrix[bottom][j] = -1
                }
            }
            bottom--
        }
        
        if left <= right {
            for i := bottom; i >= top; i-- {
                if index < len(nums) {
                    matrix[i][left] = nums[index]
                    index++
                } else {
                    matrix[i][left] = -1
                }
            }
            left++
        }
    }
    
    return matrix
}

,
                                                },
                                            },
                                        },
                                    },
                                },
                            },
                        },
                    },
                },
            },
        },
    }

    m, n := 3, 5
    result := spiralMatrix(m, n, head)
    for _, row := range result {
        fmt.Println(row)
    }
}
